common/util.go

```diff
@@ -982,3 +982,17 @@ func ConvertErrToGetTaskFailedCause(err error) types.GetTaskFailedCause {
 	}
 	return types.GetTaskFailedCauseUncategorized
 }
+
+// ConvertGetTaskFailedCauseToErr converts GetTaskFailedCause to error
+func ConvertGetTaskFailedCauseToErr(failedCause types.GetTaskFailedCause) error {
+	switch failedCause {
+	case types.GetTaskFailedCauseServiceBusy:
+		return &types.ServiceBusyError{}
+	case types.GetTaskFailedCauseTimeout:
+		return context.DeadlineExceeded
+	case types.GetTaskFailedCauseShardOwnershipLost:
+		return &types.ShardOwnershipLostError{}
+	default:
+		return &types.InternalServiceError{Message: "uncategorized error"}
```

Comment:
Do you have the message from the failedCause so we can emit here?

Reply:
The failed cause is an enum so we don't have the message here (the target cluster). The detailed error message will be logged at the source cluster during fetch. The target side only needs to know it's an unexpected error and retry.